revpi-sos

```diff
@@ -5,29 +5,44 @@
 # Copyright 2020-2023 KUNBUS GmbH
 #
 
+sos_tmp_dir=/tmp
+sos_owner=$USER
 ret=1
 force=0
 
-while getopts ":fh" OPTION; do
+while getopts ":fhd:o:" OPTION; do
 	case ${OPTION} in
 	f)
 		force=1
 		;;
+	d)
+		sos_tmp_dir="$OPTARG"
+		;;
+	o)
+		sos_owner="$OPTARG"
+		;;
 	h)
 		ret=0
 		;&
 	*)
 		echo "Usage: revpi-sos [OPTION...]"
-		echo "  -f	execute without confirmation"
-		echo "  -h	give this help list"
+		echo "	-d	DIRECTORY Specify target directory for generated report. If omitted /tmp is used"
+		echo "	-o	USER[:GROUP]	Specify name or uid of user and/or group that will be made owner of generated report."
+		echo "	-f	execute without confirmation"
+		echo "	-h	give this help list"
 		exit $ret
 		;;
 	esac
 
 done
 
+timestamp=$(date +"%s")
+sos_tmp_dir="$sos_tmp_dir/revpi-sos-$timestamp"
```

Comment:
A malicious actor could spray the `/tmp` directory with directories or symlinks and might be able to exploit this script. You should use mktemp.

Reply:
Im not sure if i know what you mean, but as this script can only be called with **sudo** so i guess the attacker probably has better ways to cause damage than through this script.

or am I missing your point?

Reply:
No that is my point. The script runs with elevated privileges. So any body with access to /tmp can exploit this!